src/windows/views/timeline_webview.py

```diff
@@ -2537,13 +2537,19 @@ def PlayheadMoved(self, position_frames):
             self.window.previewFrame(position_frames)
 
     @pyqtSlot(int)
-    def movePlayhead(self, position_frames):
+    def movePlayhead(self, position_frames, followPlayhead = False):
         """ Move the playhead since the position has changed inside OpenShot (probably due to the video player) """
-
         # Get access to timeline scope and set scale to zoom slider value (passed in)
-        code = JS_SCOPE_SELECTOR + ".MovePlayheadToFrame(" + str(position_frames) + ");"
+        code = JS_SCOPE_SELECTOR + ".MovePlayheadToFrame(%s, %s);" % (str(position_frames), "true" if followPlayhead else "false")
         self.eval_js(code)
 
+    @pyqtSlot(float)
+    def centerOnPlayhead(self):
+        """ Center the timeline on the current playhead position """
+        # Execute JavaScript to center the timeline
+        cmd = JS_SCOPE_SELECTOR + '.centerOnPlayhead();';
+        self.page().mainFrame().evaluateJavaScript(cmd)
```

Comment:
Curious, is there a reason you don't just use `self.eval_js(cmd)` here? (I know the existing code is a weird mix of using it and not using it, so there's really no right — or wrong — answer to that question.)

Reply:
No reason.  I think I originally wrote this function in a different place in the source file and the surrounding code used self.page().mainFrame().evaluateJavaScript and I didn't know any better.  self.eval_js is more concise, so I'll use it.

Reply:
The key difference, really (as I'm sure you noticed) is that `eval_js()` takes care of setting up a  delay-and-retry queue if the WebView hasn't finished loading yet. So, anything that _might_ be called during setup, before the timeline has finished loading, should always use `eval_js()`.

But that doesn't generally describe an entirely manual action such as `CenterOnPlayhead()` — unless the user is excessively twitchy and impatient, I suppose — so it's definitely a dealer's-choice sort of thing. But, as you say, `eval_js()` is _way_ more concise, and there certainly doesn't seem to be any _harm_ in using it anyway, especially for a user-triggered, infrequently-called action.